en/docs/guides/user-mgt/add-user-roles.md

```diff
@@ -0,0 +1,284 @@
+# Creating and Managing Roles
+
+Roles contain permissions for users to manage the server. You can create different roles with various combinations of permissions and assign them to a user or a group of users. You can also edit and delete an existing user role.
+
+WSO2 supports role-based authentication model where privileges of a user are based on the role it is attached to. By default, WSO2 Identity Server comes with the following roles:
+
+- Admin - Provides full access to all features and controls. By default, the user "admin" is assigned to both the "Admin" and the "Everyone" roles. The Identity Server does not allow you to configure permissions assigned to the 'Admin' role.
+- Internal/Everyone - Every new user is assigned to this role by default. It does not include any permission.
+- System - This role is not visible for modification in the interface. 
+
+If a user has several roles assigned, their permissions are added together.
+
+WSO2 products has two types of roles. **External Roles** and **Internal
+Roles**. Assume there are two user stores.
+
+<table>
+<thead>
+<tr class="header">
+<th><br />
+</th>
+<th><strong>Store-A</strong></th>
+<th><strong>Store-B</strong></th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td>Users</td>
+<td>user_A</td>
+<td>user_B</td>
+</tr>
+<tr class="even">
+<td>Roles</td>
+<td>role_A</td>
+<td>role_B</td>
+</tr>
+</tbody>
+</table>
+
+**External Roles :**  
+These roles are stored in the user stores itself. Only users that belong to a user store can be assigned to
+external roles in that user store.  
+Example : user\_A can be assigned to role\_A while user\_B cannot be assigned to role\_A.
+
+In user stores, users are assigned to **groups**. Within WSO2 IS, one user group is directly mapped to a role and permissions are assigned to the role.
+  
+**Internal Roles :**
+
+These roles are stored in the WSO2 Identity Server database. Users in all user stores can be assigned to
+these roles.
+
+In the above example, both user\_A and user\_B can be assigned to the same internal role.
+
+For internal roles, there are no mapped groups in the user stores. So we
+directly assign users to these roles.
```

Comment:
There are no mapped user groups in the user stores for internal roles. Therefore, users are directly assigned to these roles.